go/common/crypto/signature/signer.go

```diff
@@ -156,27 +162,53 @@ func SetChainContext(rawContext string) {
 type SignerRole int
 
 const (
-	SignerUnknown SignerRole = iota
-	SignerEntity
-	SignerNode
-	SignerP2P
-	SignerConsensus
-
-	// If you add to this, also add the new roles to SignerRoles.
-	maxSignerRole = SignerConsensus
+	SignerUnknown   SignerRole = 0
+	SignerEntity    SignerRole = 1
+	SignerNode      SignerRole = 2
+	SignerP2P       SignerRole = 3
+	SignerConsensus SignerRole = 4
+
+	SignerEntityName    = "entity"
+	SignerNodeName      = "node"
+	SignerP2PName       = "p2p"
+	SignerConsensusName = "consensus"
 )
 
-func (role *SignerRole) FromString(str string) error {
-	i, err := strconv.Atoi(str)
-	if err != nil {
-		return fmt.Errorf("signature: unknown signer role: '%v'", str)
+// String returns the string representation of a SignerRole.
+func (role SignerRole) String() string {
+	switch role {
+	case SignerEntity:
+		return SignerEntityName
+	case SignerNode:
+		return SignerNodeName
+	case SignerP2P:
+		return SignerP2PName
+	case SignerConsensus:
+		return SignerConsensusName
+	default:
+		return "[unknown signer role]"
 	}
+}
+
+// MarshalText encodes a SignerRole into text form.
+func (role SignerRole) MarshalText() ([]byte, error) {
```

Comment:
Is this used anywhere? Consider adding a simple round-trip test where you marshal/unmarshal all the roles.

Reply:
> Is this used anywhere?

Currently, it's not used but I added it for completeness.

> Consider adding a simple round-trip test where you marshal/unmarshal all the roles.

Done.